extensions/panache/mongodb-panache/deployment/src/test/java/io/quarkus/mongodb/panache/DuplicateIdReactiveMongoEntityTest.java

```diff
@@ -0,0 +1,54 @@
+package io.quarkus.mongodb.panache;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.platform.engine.discovery.DiscoverySelectors.selectClass;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.platform.launcher.Launcher;
+import org.junit.platform.launcher.LauncherDiscoveryRequest;
+import org.junit.platform.launcher.core.LauncherDiscoveryRequestBuilder;
+import org.junit.platform.launcher.core.LauncherFactory;
+import org.junit.platform.launcher.listeners.SummaryGeneratingListener;
+import org.junit.platform.launcher.listeners.TestExecutionSummary;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+public class DuplicateIdReactiveMongoEntityTest {
+    public static class DuplicateIdReactiveMongoEntityTestTest {
+        @RegisterExtension
+        static QuarkusUnitTest runner = new QuarkusUnitTest()
+                .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                        .addClasses(DuplicateIdReactiveMongoEntity.class))
+                .withConfigurationResource("application.properties");
+
+        @Test
+        void shouldThrow() {
+            DuplicateIdMongoEntity.findAll();
+        }
+    }
+
+    @Test
+    void verifyIdTest() {
+        // as the test will fail to be created, we need a way to launch it manually to assert
+        // that there was a build issue (the BuildException will be thrown at test instanciation time).
+        SummaryGeneratingListener listener = new SummaryGeneratingListener();
+        LauncherDiscoveryRequest request = LauncherDiscoveryRequestBuilder.request()
+                .selectors(selectClass(DuplicateIdReactiveMongoEntityTestTest.class))
+                .build();
+        Launcher launcher = LauncherFactory.create();
+        launcher.registerTestExecutionListeners(listener);
+        launcher.execute(request);
+
+        // we check that the test failed
+        TestExecutionSummary summary = listener.getSummary();
+        assertEquals(1, summary.getFailures().size());
+        String exceptionMsg = summary.getFailures().get(0).getException().getMessage();
+        assertTrue(exceptionMsg.contains("io.quarkus.builder.BuildException"));
+        assertTrue(exceptionMsg.contains("You provide a MongoDB identifier via @BsonId inside"));
+    }
```

Comment:
I wonder if in this test case we could have used this pattern instead https://github.com/quarkusio/quarkus/blob/master/extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/sql_load_script/InvalidMultilineSqlLoadScriptTestCase.java#L17 ?

Reply:
Yes, this was even documented in the extionsion guide, I need to re-read it carefully ;)

> If you want to test that an extension properly fails at build time, use the setExpectedException method:

Reply:
I updated the test with `.setExpectedException(BuildException.class)` instead of using the Launcher